### ISSUE TITLE
Sane Defaults

### ISSUE THREAD

**Issue by friedrich** [axkibe/lsyncd#481]
I do not remember ever using a program that just happened to delete any important data of mine. And I use the command line a lot. For a long time. But today it happend with lsyncd. The reason was that lsyncd behaves totally contrary to what I expected. And being a synchronization tool I expected it to behave at least remotely like rsync. That means:
- Do not delete remote files per default
- The path "example.com:" should refer to the home folder, not /
- Do not daemonize per default, so that I can see what is going on
- Respect TERM signal

I think all of these points are important. But the fact that none of them are true is catastrophic. You can guess what happened: I wanted to synchronize my home folder. "remote:" was interpreted as "remote:/". The program daemonized. I was not happy about that so I typed "killall lsyncd". Took me some time to realize that lsyncd was still running, deleting all files my remote server had access to.

**Comment by axkibe** [comment]
Hello, first I'm really sorry for the accident happened to you.

About the points:
* Term signal:
Lsyncd does respect the term signal. If it receives one, it enters the "fade" state in which it won't start any more rsync processes, but waits for the currently running ones to finish to exit in a clean state. Thats what term is IMO for.

* Daemonize by default:
Lsyncd is as it name says a daemon by design, all other unix daemons I happened to encounter so far daemonized by default. So I just tried to fall in line.

* Do not delete remote files by default:
Again, it does by default what it is designed to do, keep the target as mirror in sync with the source.

* "The path "example.com:" should refer to the home folder, not /":
Hmm this is I suppose the actual core of you accident, caused by this:
https://github.com/axkibe/lsyncd/blob/master/default-rsync.lua#L649

It is designed for targets like "hostname::syncroot" to work. When no ```/``` is appended to the target specifier, a synced directory will look like hostname::syncrootdirname which will fail. I suppose this could be improved to instead append ```/./``` so this mishap cannot happen in that way.

**Comment by friedrich** [comment]
Thanks! I think I can restore all files from the backup, so no permanent harm done.

- Term signal:
Okay, I do not have a strong opinion on that one. It just happened to make my problem worse. Maybe forwarding the signal to the child processes would be reasonable.
- Daemonize by default:
That is true for traditional SysV daemons, but not for modern ones. See: https://www.freedesktop.org/software/systemd/man/daemon.html
In any case, the important point here is that I just did not expect the program to daemonize. If you do not want to change to the new daemon style, it could be beneficial in my opinion to clearly state in the documentation and/or the usage text that the program daemonizes. Normally starting a daemon is not a big deal, so this is not important. But when synchronizing files, it can be dangerous to have the program unintentionally daemonize.
- Do not delete remote files by default:
Please write this in big red letters on the manual. rsync is also for mirroring, but does not delete per default.
- "The path "example.com:" should refer to the home folder, not /":
Yes this was the core problem. But bugs like this happen. Only the other points made it worse.

**Comment by axkibe** [comment]
Regarding the term signal: Is it documented somewhere what rsync does on a TERM signal? Finish the current file, break right away? For me it was always, "clean things up".. which most reasonable was to wait for the current children to finish.

Daemonize by default: Yes, well Lsyncd is by now over 10 years old and has been designed as very classical style daemon. With this in mind, changing default behavior is something I generally only would do very reluctantly, since I potentially anger a lot of other people, when an update on dist-upgrade suddenly changes default behavior and thus endangers their data...

Delete remote files by default: Same argument as before. Yes probably it would have been better to never do this by default. The idea never occurred to me back then as I designed Lsyncd with a specific use case in mind for which I did the whole thing the first place. The ability _not_ do delete only was added after a few years on user request. An extra notice in the manual: yes that's something I can do.

The path thing, yes will fix this "hostname:" without any directory specifier as home as default was a case never occurring to me. 

In 10 years you are the second bigger data accident at least reported to me. The other one was a user having had his home directory being the source symlinked in the target. That alone wouldn't have had mattered, but had an actual directory in the same name, which he then deleted and replaced with a symlink, which resulted the symlink being emptied. That's still something I couldn't completely rule out in the rsyncssh target, which shouldn't apply in the default rsync target tough. However, I'm always open for suggestion how to improve things...
